gazebo/rendering/LensFlare.cc

```diff
@@ -97,6 +97,12 @@ namespace gazebo
             Ogre::Vector3(static_cast<double>(this->camera->ViewportWidth()),
             static_cast<double>(this->camera->ViewportHeight()), 1.0));
 
+        if (!this->light)
```

Comment:
I think you should be able move this check to the top of the function as we don't need to set the `params` anymore

Reply:
74c0894f4a8fa971b369b18914e1748977028f3d